### PR TITLE
cluster-ui: set 'error' on eslint rule no-unused-vars

### PR DESCRIPTION
**Pull request by xinhaoz**
Unused vars will now show up as lint errors in cluster-ui package.

Epic: none

Release note: None

### CODE REVIEW

**Comment by xinhaoz** [comment]
This should be merged after https://github.com/cockroachdb/cockroach/pull/107798.

**Review by maryliag — APPROVED** [review]
:lgtm:  (after the merge of the other PR mentioned and rebase)

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/107801#-:-NaSagL11sryXUw05TAy:b-y199qb)* status: :shipit: complete! 1 of 0 LGTMs obtained

**Pull request merged**
